satpy/readers/seviri_l1b_native.py

```diff
@@ -279,20 +293,29 @@ def get_area_def(self, dataset_id):
                 [upper_area_extent, lower_area_extent,
                  upper_nlines, upper_ncols, lower_nlines, lower_ncols] = self.get_area_extent(dataset_id)
 
-                # upper area
+                # upper HRV window
                 pdict['a_desc'] = 'SEVIRI high resolution channel, upper window'
                 pdict['nlines'] = upper_nlines
                 pdict['ncols'] = upper_ncols
                 upper_area = get_area_definition(pdict, upper_area_extent)
 
-                # lower area
+                # lower HRV window
                 pdict['a_desc'] = 'SEVIRI high resolution channel, lower window'
                 pdict['nlines'] = lower_nlines
                 pdict['ncols'] = lower_ncols
                 lower_area = get_area_definition(pdict, lower_area_extent)
 
                 area = geometry.StackedAreaDefinition(lower_area, upper_area)
                 area = area.squeeze()
+
+            elif is_rapid_scan:
+                # handle RSS HRV data separately in case of padding to full disk (ncols -> 11136)
+                [lower_area_extent, lower_nlines, lower_ncols] = self.get_area_extent(dataset_id)
+                pdict['a_desc'] = 'SEVIRI high resolution channel, lower window (rss)'
+                pdict['nlines'] = lower_nlines
+                pdict['ncols'] = lower_ncols
+                area = get_area_definition(pdict, lower_area_extent)
```

Comment:
This is duplicated code from a few lines above, can you refactor it?

Reply:
you could even take in the the upper window code if you make the new method a bit more generic.

Reply:
The code segment has now been optimized and there should no longer any duplicated code.